CMakeLists.txt

```diff
@@ -89,12 +89,16 @@ endif()
 # before that it's an empty variable so we check if it's any number > 0
 if(NOT DEFINED ROOT_CXX_STANDARD)
   get_target_property(ROOT_COMPILE_FEATURES ROOT::Core INTERFACE_COMPILE_FEATURES)
-  if("cxx_std_17" IN_LIST ROOT_COMPILE_FEATURES)
+  if("cxx_std_11" IN_LIST ROOT_COMPILE_FEATURES)
+    set(ROOT_CXX_STANDARD 11)
+  elseif("cxx_std_14" IN_LIST ROOT_COMPILE_FEATURES)
+    set(ROOT_CXX_STANDARD 14)
+  elseif("cxx_std_17" IN_LIST ROOT_COMPILE_FEATURES)
     set(ROOT_CXX_STANDARD 17)
   elseif("cxx_std_20" IN_LIST ROOT_COMPILE_FEATURES)
     set(ROOT_CXX_STANDARD 20)
   else()
-    message(FATAL_ERROR "ROOT C++ could not be detected")
+    message(WARNING "ROOT c++ standard could not be detected")
```

Comment:
This I don't understand, if the standard can't be detected then that's possibly going to cause some problems later on, no? And after 11 to 20 are included this shouldn't happen anyway? Didn't ROOT change not too long ago how they define the standard? That now would build fine but with maybe different standards in ROOT and podio.

Reply:
IIUC an undefined `ROOT_CXX_STANDARD` would be caught by this:
https://github.com/AIDASoft/podio/blob/76990a98383e3519dd946d136fbd4aaa81a3c01f/CMakeLists.txt#L101-L103

Which would at least tell people what a potential problem could be rather, than a rather cryptic error about not being able to detect a c++ standard.

The `ROOT_CXX_STANDARD` is only available from 6.32 onwards. So we still need this determination. From 6.30 onwards c++17 is required. I can't remember now when they switched the default c++ standard to 17, but given that I got an email today that someone ran into this issue with 6.28, I suppose not too long ago.

Reply:
Ah yes I was thinking `cxx_std` had been changed but it was `ROOT_CXX_STANDARD`: https://github.com/AIDASoft/podio/pull/540.